examples/FetchWebsite/FetchWebsite.ino

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright 2016-2025 Hristo Gochkov, Mathieu Carbou, Emil Muratov
+
+#include <Arduino.h>
+#include <AsyncTCP.h>
+#include <StreamString.h>
+#include <WiFi.h>
+
+#include <functional>
+#include <string>
+
+void fetchAsync(const char *host, std::function<void(const StreamString *)> onDone) {
+  Serial.printf("[%s] Fetching: http://%s...\n", host, host);
+
+  // buffer where we will accumulate the received data
+  StreamString *content = new StreamString();
+
+  // reserve enough space to avoid reallocations
+  content->reserve(32 * 1024);
+
+  // create a new client
+  AsyncClient *client = new AsyncClient();
+
+  // register a callback when the client disconnects
+  client->onDisconnect([content, host, onDone](void *arg, AsyncClient *client) {
+    Serial.printf("[%s] Disconnected.\n", host);
+    onDone(content);
+    delete client;
+    delete content;
+  });
+
+  // register a callback when an error occurs
+  client->onError([host, onDone](void *arg, AsyncClient *client, int8_t error) {
+    Serial.printf("[%s] Error: %s\n", host, client->errorToString(error));
```

Comment:
this should also delete client and content. Client is gone, just like in disconnect

Reply:
i do not think so: the only place the onError callback is called is always before the onDisconnect. Same thing happens in espasycnws: onDisconnect is the only place where client is deleted. onError is just informational.

https://github.com/ESP32Async/AsyncTCP/blob/93c6ccd754ce14657dc5232394cba761e6e4a706/src/AsyncTCP.cpp#L1062-L1069

https://github.com/ESP32Async/AsyncTCP/blob/93c6ccd754ce14657dc5232394cba761e6e4a706/src/AsyncTCP.cpp#L1186-L1191

Reply:
ah ok. I seem to have forgotten that 😄